scripts/tools/metric.yml

```diff
@@ -158,6 +159,53 @@ params:
             if count is less than the number available in the raw data buffer, then driver shall only calculate that number of metric values.
 --- #--------------------------------------------------------------------------
 type: function
+desc: "Calculate one or more sets of metric values from raw data."
+version: "1.6"
+class: $tMetricGroup
+name: CalculateMultipleMetricValues
+decl: static
+details:
+    - "This function calculates more than one set of metric values from a single data buffer.  There may be one set of metric values for each sub-device, for example."
+    - "Each set of metric values may consist of a different number of metric values, returned as the metric value count."
+    - "All metric values are calculated into a single buffer; use the metric counts to determine which metric values belong to which set."
+    - "The application may call this function from simultaneous threads."
+params:
+    - type: "$t_metric_group_handle_t"
+      name: hMetricGroup
+      desc: "[in] handle of the metric group"
+    - type: "$t_metric_group_calculation_type_t"
+      name: type
+      desc: "[in] calculation type to be applied on raw data"
+    - type: size_t
+      name: rawDataSize
+      desc: "[in] size in bytes of raw data buffer"
+    - type: "const uint8_t*"
+      name: pRawData
+      desc: "[in][range(0, rawDataSize)] buffer of raw data to calculate"
+    - type: uint32_t*
+      name: pSetCount
+      desc: |
+            [in,out] pointer to number of metric sets.
+            if *pSetCount is zero, then the driver shall update the value with the total number of metric sets to be calculated.
+            if *pSetCount is greater than the number available in the raw data buffer, then the driver shall update the value with the actual number of metric sets to be calculated.
+            if *pSetCount is is less than the number available in the raw data buffer, then the driver shall calculate the metrics for up to that number of sub-devices.
+    - type: uint32_t*
+      name: pTotalMetricValueCount
+      desc: |
+            [in,out] pointer to number of the total number of metric values calculated, for all metric sets.
+            if *pSetCount is zero, then the driver shall update the value with the total number of metric values to be calculated.
+            if *pSetCount is greater than the number available in the raw data buffer, then the driver shall update the value with the actual number of metric values to be calculated.
+    - type: uint32_t*
+      name: pMetricCounts
+      desc: |
+            [in,out][optional][range(0, *pSetCount)] buffer of metric counts per metric set.
+    - type: "$t_typed_value_t*"
+      name: pMetricValues
+      desc: |
+            [in,out][optional][range(0, *pTotalMetricValueCount)] buffer of calculated metrics.
+            if *pSetCount is less than the number available in the raw data buffer, then driver shall only calculate that number of metric values.
```

Comment:
should pSetCount be pTotalMetricValueCount

Reply:
good catch

Reply:
@matcabral this is the same as the experimental extension. @matcabral please confirm then what it should go here.

Reply:
@matcabral 
Implementation updates "pMetricValues"  for only for the "psetCount" provided.

Reply:
@matcabral , @joshuaranjan : could you suggest how the parameters should be named and described? in this interface, we have two count variables.

pSetCount is associated with pMetricCounts, could we suggest a different name for pMetricCounts to not have the "Count" in it?

pTotalMetricValueCount is associated with pMetricValues. So pTotalMetricValueCount  could be just called pMetricValueCount?

Reply:
@jandres742 In respect to the names, I think the current ones are accurate.  In terms of the description, I have some comments below:

>pSetCount is associated with pMetricCounts, could we suggest a different name for pMetricCounts to not have the "Count" in it?

An actual clearer name would be `pMetricCountPerSet` : Note that this is an array of sizes/counts, with each element being the number of metrics available in each `set` (defied by `pSetCount`). Each set can have different numbers. 

> pTotalMetricValueCount is associated with pMetricValues. So pTotalMetricValueCount could be just called pMetricValueCount?

Strictly speaking, your statement is correct. However, since there are few counts in the API the intention was to be upfront clear. I don’t have a strong opinion about either name. Maybe improving a more explicit description would allow changing :

` pointer to number of the total number of metric values calculated, it is the sum of all elements in buffer  pMetricCountPerSet . if count is zero...`

It is worth to note that in the original version of the API (zetMetricGroupCalculateMetricValues()) the parameter is called pMetricValueCount, so keeping the same name could be seen favorable. 


In terms of the description and behavior of description and behavior :

I saw that the pSetCount description missed a text describing it can be used to limit the number of sets to be calculated. So the following text should be appended at the end:

` if count is less than the number of available sets in the raw data buffer, then driver shall only calculate that number of sets. ` 

Still need to define how the user can limit the number of metrics to be calculated, let me drive that discussion by email and write the resolution here.

Reply:
+1 for "pMetricCountPerSet" as suggested above

>if count is less than the number of available sets in the raw data buffer, then driver shall only calculate that number of sets.
@matcabral , In explanation about "count" in other instances, the spec covers 2 conditions
1. what happens when it is 0
2. what happens when count is greater than available.

Since both of these are covered for "pSetCount ", please share if we need the additional point that is suggested ?
Thanks

Reply:
> Since both of these are covered for "pSetCount ", please share if we need the additional point that is suggested ?

@joshuaranjan   yes:   

1.  0 is a query
2. greater than available, scopes to whats available 
3.  smaller than avaialble, limits to that number